File: Tests/test10/equi_filter.pl
#!/usr/bin/perl -w

use strict;

my %letterCount = ();
my @equiWord = ();
my $answer = "";

foreach my $line (<STDIN>) {

	# remove extra spaces
	$line =~ s/\s+/ /g;
	# split line into words, space delimited
	my @words = split / /, $line;

	foreach my $word (@words) {

		# remove the trailing newline
		$word =~ s/\n//g;

		# split word into letters
		my @letters = split //, $word;

		foreach my $letter (@letters) {
			# increase count of lc(letter) in hash
			$letterCount{lc($letter)} += 1;
		}

		# variable to store count of individual letter
		my $currCount = 0;
		# flag to determine if a word is an equi-word
		my $equiWord = 1;
		# foreach each letter's count
		foreach my $count (keys %letterCount) {
			# if the current count is zero or not changed
			if ($currCount == 0 || $currCount == $letterCount{$count}) {
				# set current count to this letter's count
				$currCount = $letterCount{$count};
			} else {
				# word is not an equi-word
				$equiWord = 0;
				# break the loop
				last;
			}
		}
		# if word is an equi-word, print it
		if ($equiWord == 1) {
			print "$word ";
		}

		# clear all keys in the hash
		foreach my $count (keys %letterCount) {
			delete $letterCount{$count};
		}
	}

	print "\n";

}

File: Tests/test10/equi_filter.py
#!/usr/bin/python

import sys, re

letterCount = {}
equiWord = []

for line in sys.stdin:
	line = re.sub("\s+"," ",line)

	words = line.split(' ')

	for word in words:
		#word = re.sub("\n","",word)

		letters = list(word)

		for letter in letters:
			#print ("letter %s"%(letter))
			if letter in letterCount:
				letterCount[letter.lower()] += 1
			else:
				letterCount[letter.lower()] = 1

		currCount = 0
		equiWord = 1
		for letter in letterCount:
			if (currCount == 0 or currCount == letterCount[letter]):
				currCount = letterCount[letter]
			else:
				equiWord = 0
				break

		if (equiWord == 1):
			if (word == "Not-equi:"):
				print "\n", word,
			else:
				print word,

		letterCount.clear()


	
